extensions/sql-database-projects/src/models/project.ts

```diff
@@ -2140,6 +2144,38 @@ export class Project implements ISqlProject {
 
 		return folderEntry;
 	}
+
+	/**
+	 * Moves a file to a different location
```

Comment:
Can you expand this to say that an `undefined` result means it was a no-op?  Or maybe we can get rid of the `undefined` output by returning success in the event of a no-op (and maybe emit telemetry if there's an unhandled node type that's moved)?  Not married to it, just an idea.

Reply:
updated so that a ResultStatus is always returned and now an error shows if attempting to move an unhandled node